server/src/main/java/org/elasticsearch/index/mapper/BlockDocValuesReader.java

```diff
@@ -358,11 +360,47 @@ public String toString() {
 
     /**
      * Convert from the stored {@link long} into the {@link double} to load.
-     * Sadly, this will go megamorphic pretty quickly and slow us down,
-     * but it gets the job done for now.
      */
-    public interface ToDouble {
+    public sealed interface ToDouble permits SortableLongToDouble, SortableIntToFloat, SortableShortToHalfFloat, LongToScaledFloat {
```

Comment:
NIT: I do not think we need `permits` when all implementations are in the same file, do we?